docs/docs/reference-react.md

```diff
@@ -0,0 +1,149 @@
+
+## Reference
+
+ - [`React.Component`](#react.component)
+ - [`createClass()`](#createclass)
+ - [`createElement()`](#createlement)
+ - [`cloneElement()`](#cloneelement)
+ - [`createFactory()`](#createfactory)
+ - [`isValidElement()`](#isvalidelement)
+ - [`PropTypes`](#proptypes)
+ - [`Children`](#children)
+
+### `React.Component`
+
+```javascript
+class Greeting extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+`React.Component` is the base class for React components when they are defined using ES6 classes. See the [React.Component API Reference](/react/docs/reference-react-component.html) for a list of methods are provided by the base `React.Component` class.
+
+### `createClass()`
+
+```javascript
+React.createClass(specification)
+```
+
+If you don't use ES6 yet, you may use the `React.createClass` helper instead to create a component class. See [Using React without ES6](/react/docs/react-without-es6.html) for more information.
+
+### `createElement()`
+
+```javascript
+React.createElement(
+  type,
+  [props],
+  [children ...]
+)
+```
+
+Create and return a new React element of the given type. The type argument can be either an
+html tag name string (eg. 'div', 'span', etc), or a React class (created via `React.createClass`).
+
+Code written with JSX will be converted to use `React.createElement`. You will not typically call `React.createElement` directly if you are using JSX.
+
+Convenience wrappers around `React.createElement` for DOM components are provided by `React.DOM`. For example, `React.DOM.a(...)` is a convenience wrapper for `React.createElement('a', ...)`.
+
+See [React Without JSX](/react/docs/react-without-jsx.html) to learn more.
+
+### `cloneElement()`
+
+```
+React.cloneElement(
+  element,
+  [props],
+  [children ...]
+)
+```
+
+Clone and return a new React element using `element` as the starting point. The resulting element will have the original element's props with the new props merged in shallowly. New children will replace existing children. `key` and `ref` from the original element will be preserved.
+
+`React.cloneElement` is almost equivalent to:
+
+```js
+<element.type {...element.props} {...props}>{children}</element.type>
+```
+
+However, it also preserves `ref`s. This means that if you get a child with a `ref` on it, you won't accidentally steal it from your ancestor. You will get the same `ref` attached to your new element.
+
+This API was introduced as a replacement of the deprecated `React.addons.cloneWithProps`.
+
+### `createFactory()`
+
+```javascript
+React.createFactory(type)
+```
+
+Return a function that produces React elements of a given type. Like [`React.createElement`](#createElement), the type argument can be either an html tag name string (eg. 'div', 'span', etc), or a `ReactClass`.
+
```

Comment:
I'd mention you don't need createFactory in JSX code. It mostly exists for people who don't use JSX.